README.md

```diff
@@ -1 +1,96 @@
-# Reloader
\ No newline at end of file
+# RELOADER
+
+## WHY NAME RELOADER
+
+In english language, Reloader is a thing/tool that can reload certain stuff. So refereig to that meaning relaoder can reload
+
+## Problem
+
+We would like to watch if some change happens in `ConfigMap` and `Secret` objects and then perform certain upgrade on relavent `Deployment`, `Deamonset` and `Statefulset`
+
+## Solution
+
+Reloader can watch any changes in `ConfigMap` and `Secret` objects and then performs rolling upgrades on their associated `Deployments`, `Deamonsets` and `Statefulsets` and updating these dynamically.  
+
+**NOTE:** This controller has been inspired from [configmapController](https://github.com/fabric8io/configmapcontroller)
+
+## How to use Reloader
+
+For a `Deployment` called `foo` have a `ConfigMap` called `foo`. Then add this annotation to your `Deployment`
+
+```yaml
+metadata:
+  annotations:
+    reloader.stakater.com/update-on-change: "foo"
+```
+
+Then, providing `Reloader` is running, whenever you edit the `ConfigMap` called `foo` the Reloader will update the `Deployment` by adding the environment variable:
+
+```
+STAKATER_FOO_REVISION=${reloaderRevision}
```

Comment:
@faizanahmad055 @waseem-h Is the whole config needed in the environment variable? Or does it just need an environment variable that changes value? As I've faced an issue once when the revision value gets to long and kubectl apply fails to update it.

Reply:
It's just used for comparing old and new configmap. We can replace it with something like sha1 or md5